pjlib/src/pj/sock_bsd.c

```diff
@@ -905,4 +905,127 @@ PJ_DEF(pj_status_t) pj_sock_accept( pj_sock_t serverfd,
 }
 #endif	/* PJ_HAS_TCP */
 
+/*
+ * Simulate unix-like socketpair()
+ * Use a pair of IPv4/IPv6 local sockets (listening on 127.0.0.1/::1)
+ */
+static int socketpair_s(int family, int type, int protocol, pj_sock_t sv[2])
+{
+    int status;
+    pj_sock_t lfd = PJ_INVALID_SOCKET;
+    pj_sock_t cfd = PJ_INVALID_SOCKET;
+    pj_str_t loopback;
+    pj_sockaddr sa;
+    int salen;
+
+#if PJ_HAS_TCP
+    PJ_ASSERT_RETURN(type == pj_SOCK_DGRAM() || type == pj_SOCK_STREAM(),
+		     PJ_EINVAL);
+#else
+    PJ_ASSERT_RETURN(type == pj_SOCK_DGRAM(), PJ_EINVAL);
+#endif
+
+    PJ_ASSERT_RETURN(family == pj_AF_UNIX() || family == pj_AF_INET() ||
+			 family == pj_AF_INET6(), PJ_EINVAL);
```

Comment:
I'd say it's better to reject pj_AF_UNIX() altogether, rather than silently changing the type to pj_AF_INET() below.

Reply:
For compatibility with unix-like socketpair(), must support AF_UNIX，so  when family is pj_AF_UNIX(), change it to pj_AF_INET()

Reply:
Perhaps it would be clearer if we write down the spec of the function:

- On Linux/Unix, `af` values of AF_UNIX, AF_INET, and AF_INET6 are supported
- On Windows/others that don't have AF_UNIX, only `af` values AF_INET and AF_INET6 are supported
- In all cases, the correct type of socket must be returned according to `af` and `type`

Reply:
I have removed  forcely convert AF_UNIX to AF_INET logic